e2e/cypress/integration/messaging/focus_move_spec.js

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// ***************************************************************
+// - [#] indicates a test step (e.g. 1. Go to a page)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+import * as TIMEOUTS from '../../fixtures/timeouts';
+
+describe('Messaging', () => {
+    before(() => {
+        cy.apiLogin('user-1');
+        cy.visit('/');
+    });
+
+    it('M15406 - Focus move from Recent Mentions to main input box when a character key is selected', () => {
+        //#Click the flag icon to open the flagged posts RHS to move the focus out of the main input box
+        cy.get('#channelHeaderFlagButton').click();
+
+        //#Making sure Flagged Posts is present on the page
+        cy.contains('Flagged Posts');
+
+        //#Push a character key such as "A"
+        cy.get('#post_textbox', {timeout: TIMEOUTS.LARGE}).clear().type('A');
```

Comment:
Typing directly to the post text box sorta defeats the purpose of this test as `.type()` will first click on the desired element first to focus it before typing - https://docs.cypress.io/api/commands/type.html#Events.

You may want do do `cy.get('body').type('A')` instead.

Reply:
Yeah, this is tricky.  However, I'm not sure if Cypress will allow typing if out of focus.

Reply:
They do, https://docs.cypress.io/api/commands/type.html#Global-Shortcuts. Depends on where we are listening for keys I suppose.

Reply:
![Screen Shot 2019-07-29 at 4 45 25 PM](https://user-images.githubusercontent.com/10777374/62090065-5a962e00-b220-11e9-92f1-cfc603668517.png)

I focus on the div that contains the create post input and I got this message, not sure if we can do that.

Reply:
what about adding tabIndex to a div parent https://docs.cypress.io/api/commands/type.html#Tabindex  @saturninoabril @thekiiingbob ?

Reply:
I'm on the fence of doing that but it would be interesting if you get it working. On the other hand, Cypress docs states that `body` allows to type into. It actually is, only that it's not working as we expect it to be.

@thekiiingbob Any thoughts?

If no workaround, I'm okay to mark this as Cypress limitation.

Reply:
Alrighty, played around with it for a bit. The focus does in fact move to the textarea, but for some reason the keydown event does not ultimately trigger on the textarea. So, something like this could be sufficient enough I think.

```js
t('M15406 - Focus move from Recent Mentions to main input box when a character key is selected', () => {
        //#Click the flag icon to open the flagged posts RHS to move the focus out of the main input box
        cy.get('#channelHeaderFlagButton').click();

        //#Making sure Flagged Posts is present on the page
        cy.contains('Flagged Posts').should('be.visible');
        cy.get('#post_textbox').should('not.be.focused');

        //#Push a character key such as "A"
        cy.get('body').type('A');

        //#Expect to have "A" value in main input
        cy.get('#post_textbox').should('be.focused');

        //#Click the @ icon to open the Recent mentions RHS to move the focus out of the main input box
        cy.get('#channelHeaderMentionButton').click();
        cy.get('#post_textbox').should('not.be.focused');

        //#Push a character key such as "B"
        cy.get('body').type('B');
        cy.get('#post_textbox').should('be.focused');
    });
```

Doesn't validate that the text is actually sent, but at least we know the focus is getting moved?

Reply:
@thekiiingbob  yea I also tried to focus on a div with tabindex property and is the same result, I think this would be the only way to test this, I updated the PR with your code.


@thekiiingbob @saturninoabril  Please merge this PR if you think is ok, I think is good enough for the requirement.